sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/EncryptingData.cs

```diff
@@ -0,0 +1,107 @@
+
+            RSACryptoServiceProvider myRSAC = new RSACryptoServiceProvider();
+            // Review the use of any TryEncrypt/TryDecrypt and specific Encrypt/Decrypt of RSA subclasses.
+            myRSAC.Encrypt(data, false);    // Noncompliant
+            myRSAC.Decrypt(data, false);    // Noncompliant
+            int written;
+
+            // Note: TryEncrypt/TryDecrypt are only in .NET Core 2.1+
+            //            myRSAC.TryEncrypt(data, Span<byte>.Empty, padding, out written); // Non compliant
+            //            myRSAC.TryDecrypt(data, Span<byte>.Empty, padding, out written); // Non compliant
+
+            byte[] rgbKey = { 1, 2, 3 };
+            byte[] rgbIV = { 4, 5, 6 };
+            SymmetricAlgorithm rijn = SymmetricAlgorithm.Create();
+            // Review the creation of Encryptors from any SymmetricAlgorithm instance.
+            rijn.CreateEncryptor();
+//          ^^^^^^^^^^^^^^^^^^^^^^ {{Make sure that encrypting data is safe here.}}
+            rijn.CreateEncryptor(rgbKey, rgbIV);    // Noncompliant
+            rijn.CreateDecryptor();                 // Noncompliant
+            rijn.CreateDecryptor(rgbKey, rgbIV);    // Noncompliant
+        }
+
+    }
+    public class MyAsymmetricCrypto : System.Security.Cryptography.AsymmetricAlgorithm // Noncompliant
+    {
+        // ...
+    }
+
+    public class MySymmetricCrypto : System.Security.Cryptography.SymmetricAlgorithm // Noncompliant 
+    {
+        public override ICryptoTransform CreateDecryptor(byte[] rgbKey, byte[] rgbIV) { return null; }
+        public override ICryptoTransform CreateEncryptor(byte[] rgbKey, byte[] rgbIV) { return null; }
+        public override void GenerateIV() { /* no-op */ }
+        public override void GenerateKey() { /* no-op */ }
+    }
+
+
+    public class Class2
+    {
+        public void AdditionalTests(Byte[] data, RSAEncryptionPadding padding)
+        {
+            var customAsymProvider = new MyRSA();
+
+            // Should raise on derived asymmetric classes
+            customAsymProvider.Encrypt(data, padding);  // Noncompliant
+            customAsymProvider.EncryptValue(data);      // Noncompliant
+            customAsymProvider.Decrypt(data, padding);  // Noncompliant
+            customAsymProvider.DecryptValue(data);      // Noncompliant
+
+            // Should raise on the Try* methods added in NET Core 2.1
+            // Note: this test is cheating - we can't currently referencing the
+            // real 2.1 assemblies since the test project is targetting an older
+            // NET Framework, so we're testing against a custom subclass
+            // to which we've added the new method names.
+            customAsymProvider.TryEncrypt();        // Noncompliant
+            customAsymProvider.TryEncrypt(null);    // Noncompliant
+            customAsymProvider.TryDecrypt();        // Noncompliant
+            customAsymProvider.TryDecrypt(null);    // Noncompliant
```

Comment:
This is the hacky test workaround. I spent a bit of time trying to work out if we could easily change the framework version used in each test, but could find a simple way to do it.